Source/WebCore/css/parser/CSSPropertyParserConsumer+Animations.cpp

```diff
@@ -37,6 +37,48 @@
 namespace WebCore {
 namespace CSSPropertyParserHelpers {
 
+Vector<std::pair<CSSValueID, double>> consumeKeyframeKeyList(CSSParserTokenRange& range)
+{
+    auto timelineRange = [](CSSParserTokenRange& range, CSSValueID id) -> std::optional<std::pair<CSSValueID, double>> {
+        if (CSSPropertyParserHelpers::isAnimationRangeKeyword(id)) {
+            // "normal" will be considered valid by isAnimationRangeKeyword() but is not valid for a @keyframes rule.
+            if (id == CSSValueNormal)
+                return { };
+            auto& token = range.consumeIncludingWhitespace();
+            if (token.type() == PercentageToken && token.numericValue() >= 0 && token.numericValue() <= 100)
```

Comment:
Assuming this is for https://drafts.csswg.org/scroll-animations-1/#named-range-keyframes, these are supposed to be `<percentage [0,100]>` which include calc(), so should either use `consumePercentage()` or `MetaConsumer<CSS::Percentage<CSS::Range{0,100}>>`.

If the spec intends for calc() not to be included, it can use the production `<percentage-token>` (https://drafts.csswg.org/css-values-5/#typedef-progress is an example of `<percentage-token>` being used).

Reply:
Thanks for the tip! Using `consumePercentage()` and `CSSPrimitiveValue::resolveAsPercentage()` made an additional WPT test pass which tested a `calc(20% + 30%)` value which previously caused a parsing error.